cleanlab/multiannotator.py

```diff
@@ -0,0 +1,928 @@
+    num_annotations : np.ndarray
+        An array of shape ``(N,)`` with the number of annotators that have labeled each example.
+    annotator_agreement : np.ndarray
+        An array of shape ``(N,)`` with the fraction of annotators that agree with each consensus label.
+    model_weight : float
+        float specifying the model weight used in weighted averages,
+        None if model weight is not used to compute quality scores
+    annotator_weight : np.ndarray
+        An array of shape ``(M,)`` where M is the number of annotators, specifying the annotator weights used in weighted averages,
+        None if annotator weights are not used to compute quality scores
+    consensus_quality_score : np.ndarray
+        An array of shape ``(N,)`` with the quality score of the consensus.
+    quality_method : str, default = "auto" (Options: ["auto", "agreement"])
+        Specifies the method used to calculate the quality of the consensus label.
+        For valid quality methods, view :py:func:`get_label_quality_multiannotator <cleanlab.multiannotator.get_label_quality_multiannotator>`
+
+    Returns
+    -------
+    annotator_stats : pd.DataFrame
+        Returns overall statistics about each annotator.
+        For details, see the documentation of :py:func:`get_label_quality_multiannotator <cleanlab.multiannotator.get_label_quality_multiannotator>`.
+    """
+
+    annotator_quality = _get_annotator_quality(
+        labels_multiannotator=labels_multiannotator,
+        pred_probs=pred_probs,
+        consensus_label=consensus_label,
+        num_annotations=num_annotations,
+        annotator_agreement=annotator_agreement,
+        model_weight=model_weight,
+        annotator_weight=annotator_weight,
+        quality_method=quality_method,
+    )
+
+    # Compute the number of labels labeled/ by each annotator
+    num_labeled = labels_multiannotator.count()
+
+    # Compute the fraction of labels annotated by each annotator that agrees with the consensus label
+    # TODO: check if we should drop singleton labels here
+    agreement_with_consensus = labels_multiannotator.apply(
+        lambda s: np.mean(s[pd.notna(s)] == consensus_label[pd.notna(s)]),
+        axis=0,
+    ).to_numpy()
+
+    # Find the worst labeled class for each annotator
+    # TODO: potentially have to do tiebreaking here
+    worst_class = labels_multiannotator.apply(
+        lambda s: (s[pd.notna(s)] == consensus_label[pd.notna(s)]).groupby(s).mean().idxmin()
+    ).astype("int64")
+
+    # Create multi-annotator stats DataFrame from its columns
+    annotator_stats = pd.DataFrame(
+        {
+            "annotator_quality": annotator_quality,
+            "num_labeled": num_labeled,
+            "agreement_with_consensus": agreement_with_consensus,
+            "worst_class": worst_class,
+        }
+    )
+
+    return annotator_stats.sort_values(by=["annotator_quality", "agreement_with_consensus"])
+
+
```

Comment:
Let's move `get_label_quality_multiannotator` to the top of this file since it's the most important function.